wicker/core/datasets.py

```diff
@@ -187,19 +269,36 @@ def _get_dataset_size(self):
                 bucket, key = path.replace("s3://", "").split("/", 1)
                 buckets_keys.add((bucket, key))
 
-        # we use existing shuffle worker here to grab the column file from
-        # know location.
-        def get_file_size_s3(bucket_key: Tuple[str, str]) -> int:
-            s3 = boto3.resource("s3")
-            bucket, key = bucket_key
-            bucket, key = path.replace("s3://", "").split("/", 1)
-            byte_length = s3.Object(bucket, key).content_length
-            return byte_length
+        # declare the variables neccessary to split
+        # out between multiple procs
+        buckets_keys_chunks = []
+        manager = Manager()
+        total_size = manager.Value("i", 0)
+        lock = manager.Lock()
+        buckets_keys_list = list(buckets_keys)
+        chunk_size = 500
```

Comment:
@pickles-bread-and-butter let's do this - you computed `buckets_keys` at this point, now you are going to do chunking and setup a `Pool` for multiprocessing. Can we move this last part (chunking and setting up the pool computation) into its own helper function (that is called by `_get_dataset_size`) and then move this new function up closer to your new `get_file_size_s3` function?

Then note that both your new function and `get_file_size_s3` do chunking and pooling operations. We *might* be able to condense them further or at least making reading them super natural. I would put the new function first (above `get_file_size_s3`) so that when you read the code your eyes just go from the Pooling operation and `map` call to `get_file_size_s3` immediately to the `get_file_size_s3` function next.